Mathlib/NumberTheory/Transcendental/Lindemann/Basic.lean

```diff
@@ -205,8 +206,20 @@ private theorem linearIndependent_exp' (u : ι → ℂ) (hu : ∀ i, IsIntegral
 
 theorem linearIndependent_exp (u : ι → integralClosure ℚ ℂ) (u_inj : u.Injective) :
     LinearIndependent (integralClosure ℚ ℂ) fun i ↦ exp (u i) :=
-  Fintype.linearIndependent_iff.mpr fun v h ↦ mod_cast
-    linearIndependent_exp' (u ·) (u · |>.2) (fun i j ↦ by simpa using @u_inj i j) (v ·) (v · |>.2) h
+  linearIndependent_iff'.mpr fun s v h ↦ by
+    simpa using linearIndependent_exp' (ι := s) (u ·) (u · |>.2)
+      (fun i j ↦ by simpa [Subtype.coe_inj] using @u_inj i j)
+      (v ·) (v · |>.2) (by simpa [sum_attach _ fun x ↦ v x * cexp (u x)])
+
+theorem algebraicIndependent_exp (u : ι → integralClosure ℚ ℂ) (hu : LinearIndependent ℕ u) :
+    AlgebraicIndependent (integralClosure ℚ ℂ) fun i ↦ exp (u i) := by
```

Comment:
AlgebraicIndependent over `integralClosure ℚ ℂ` is the same as AlgebraicIndependent over ℚ (or ℤ), right? Do we have this? @acmepjz

Reply:
Seems that we don't have this yet. Let me check later.

Reply:
> Looks like [Algebra.IsAlgebraic.trans](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Algebraic.html#Algebra.IsAlgebraic.trans) and [IntermediateField.isAlgebraic_adjoin_iff_isAlgebraic](https://leanprover-community.github.io/mathlib4_docs/Mathlib/FieldTheory/AlgebraicClosure.html#IntermediateField.isAlgebraic_adjoin_iff_isAlgebraic) are two things to be generalized.

I've generalized these to domains and proven the Transcendental version of this [here](https://github.com/leanprover-community/mathlib4/compare/Subalgebra.algebraicClosure?expand=1#diff-1fed4ce28689dbfee7313dcc400e1b532d2bc3e97b932d74791f8506e07f9083R932-R939) but I'll wait for #6183 and #19370 before making a PR.

Reply:
> By the way, [Transcendental.of_tower_top_of_injective](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Algebraic.html#Transcendental.of_tower_top_of_injective) should be renamed to `Transcendental.restrictScalars` and [IsAlgebraic.tower_top_of_injective](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Algebraic.html#IsAlgebraic.tower_top_of_injective) should be renamed to `IsAlgebraic.extendScalars`, as someone already pointed out that the original naming is not appropriate.

Why were `IsAlgebraic.tower_top_of_subalgebra_le` and `Transcendental.of_tower_top_of_subalgebra_le` not changed to `extend/restrictScalars`?

Reply:
I've proven [`AlgebraicIndependent.extendScalars`](https://github.com/leanprover-community/mathlib4/compare/Subalgebra.algebraicClosure?expand=1#diff-79fb72674b5880e7c904ddfcdbb728fd400ab58285267410bccebb4b118337baR627-R628); will merge master and clean up when I get up.

Reply:
@alreadydone I don't know what you want to change here - feel free to push directly to this PR

Reply:
Now it suffices to show `AlgebraicIndependent ℚ fun i ↦ exp (u i)` but that may not make the proof easier ...